framework/src/controller/channels/ipc_channel.ts

```diff
@@ -162,37 +191,43 @@ export class IPCChannel extends BaseChannel {
 			return handler(action.params) as T;
 		}
 
-		return new Promise((resolve, reject) => {
-			this._rpcClient.call(
-				'invoke',
-				action.toJSONRPCRequest(),
-				(err: JSONRPC.JSONRPCError | undefined, res: JSONRPC.ResponseObjectWithResult<T>) => {
-					if (err) {
-						return reject(err);
-					}
-					return resolve(res.result);
-				},
-			);
+		// When the handler is in other channels
+		return new Promise(resolve => {
+			// Subscribe to the action Id;
+			this._rpcRequestIds.add(action.id as string);
+			this._subSocket.subscribe(action.id as string);
+			this._rpcClient.send(['invoke', JSON.stringify(action.toJSONRPCRequest())])
+				.then(_ => {
+					this._emitter.once(action.id as string, (response: JSONRPC.ResponseObjectWithResult<T>) => {
```

Comment:
What if we don't received the response for that `action.id`?

Reply:
we can add a timeout